### DISCUSSION TITLE
Switch auth for different endpoints

### DISCUSSION THREAD

**Discussion by futchitwo** [deta/cloud-docs#404] [Help]
Is it possible to switch between authentication and non-authentication for different api endpoints within a single micro? For example, I would like to define a public API and an administrative API in the same micro.

**Comment by jnsougata** [comment]
Ig utilizing headers will give you the functionality that you are looking for
if header has special auth in it ... it will go to admin mode otherwise public mode something like this

**Comment by lemonyte** [comment]
Using the Micro's settings, no. The micro runner can add authentication for the root domain of your micro (`<name>.deta.dev`), and as a result, also applies to any endpoint. Endpoints are handled by your own code only, so you need to implement authentication on specific endpoints yourself, be it through headers, api, access token, or anything else you can think of. If I remember correctly, the Deta team is working on adding another service to the stack, called Deta Auth, which may or may not help you. Can't say how soon this will come.
